pingouin/multivariate.py

```diff
@@ -254,3 +254,102 @@ def multivariate_ttest(X, Y=None, paired=False):
     stats = {'T2': t2, 'F': fval, 'df1': df1, 'df2': df2, 'pval': pval}
     stats = pd.DataFrame(stats, index=['hotelling'])
     return _postprocess_dataframe(stats)
+
+def box_m (data, vars=None, group=None, alpha=.001):
+    from scipy.stats import chi2
+    """Test equality of covariance matrices.
+
+    Parameters
+    ----------
+    data : :py:class:`pandas.DataFrame`, list or dict
+        Iterable. Can be either a list / dictionnary of iterables
+        should be a long-format pandas dataframe.
+    vars : str
+        Variables of covariance matrix (only when ``data`` is a long-format dataframe).
+    group : str
+        Grouping variable (only when ``data`` is a long-format dataframe).
+    alpha : float
+        Significance level.
+
+    Returns
+    -------
+    stats : :py:class:`pandas.DataFrame`
+
+        * ``'Chi2'``: Test statistic 
+        * ``'pval'``: p-value
+        * ``'df'``: The Chi-Square statistic's degree of freedom
+        * ``'equal_cov'``: True if ``data`` has equal covariance
+        
+    Notes
+    -----
+    Pooled sample covariance matrix :math:`S_{\\text{pl}}` is calculated as
+    .. math::
+
+        S_{\\text{pl}} = \\frac{\\sum_{i=1}^k(n_i-1)\\textbf{S}_i}{\\sum_{i=1}^k(n_i-1)},
+    where :math:`n_i` and :math:`S_i` are the sample size and covariance matrix of the :math:`i^{th}` sample, :math:`k` is the number of independent samples.
+    More mathematical expressions can be found in [1]
+    
+    References
+    ----------
+    .. [1] Rencher, A. C. (2003). Methods of multivariate analysis (Vol. 492). John Wiley & Sons.
+
+
+    Examples
+    --------
+    1. Box's M testing 3 covariance matrices from 'tip' dataset
+
+    >>> import numpy as np
+    >>> import pingouin as pg
+    >>> import scipy.stats
+    >>> data = pg.read_dataset('tips')[['total_bill','tip','size']]
+
+    >>> box_m(data,vars=['total_bill','tip'],group='size')
+
+            Chi2	df	    pval	    equal_cov
+    box	45.842377	15.0	0.000056	False
+
+    """
+    
+    assert isinstance(data, (pd.DataFrame, list, dict))
+    if isinstance(data, pd.DataFrame):
+        assert vars is not None and group is not None, 'data should be in the long-format'
+        # Long-format Pandas DataFrame
+        assert group in data.columns
+        assert set(vars).issubset(data.columns)
+        grp = data.groupby(group, observed=True)[vars]
+        assert grp.ngroups > 1, 'Data must have at least two columns.'
+        covs = grp.cov()
+        num_covs,num_vars = covs.index.levshape
+        sizes = grp.count().iloc[:,0]        
+
```

Comment:
Please avoid double blank lines. Can you please check that the code follows the flake8 guidelines? Thanks!

Reply:
Could you explain more in detail how to run flake8 checks? The instruction on the contribution page is not so clear.

Reply:
You can install the flake8 package and then analyze your code with `flake8` command. See more details here: https://pypi.org/project/flake8/